### DISCUSSION TITLE
How to set correct Canvas Size and Merge Parameters

### DISCUSSION THREAD

**Discussion by Mevel** [photobooth-app/photobooth-app#350] [Howtos and camera specific topics]
Hey Guys!  I need a little help to get my Booth running smooth.  Pi5 with a webcam 1920 x 1080 resolution  I designed a overlay Frame optimal for printing on 148mm X 100mm Postcard Paper.  Problem: whenever i use the Merge Definitions inside the collage feature and resize the images taken, they get cropped very hard the left and right part is "missing".  exactly the ~100px left and right comapred to 1920 camera capture vs 1741px output to canvas size.  althou its says aspect is kept in the merge definition  mediaprocessing: full still width 1748px  collage canvas size: 1748px frame size: 1748px  any resized image is cropped....  any idea where i am going wrong?  Thanks

**Comment by mgineer85** [comment]
Hey, I am not sure if I exactly understand what you want to achieve. 

In the end I think you have different aspect ratio for print vs webcam. So there needs to be some kind of adjustment to fit the captures to the merged collage.
To do that you could squeeze the capture ignoring the aspect ratio. The image would look distorted but nothing is cropped.
Since distorted images are not acceptable I chose to crop the image to make it fit in the given box. In that equation height and with are checked.

I hope this helps you to better understand what is going on under the hood.

If you think there is a mistake maybe you could add sample images depicting the issue?

regards

**Reply by Mevel** [reply]
Hey Michael, danke für die schnelle Antwort!


I will try to explain a bit more in detail.

If I use the merge definition in the collage settings, the images don't get resized to fit into the given aspect ratio set in the parameters.

The images that should get resized are cropped to the given pixel dimensions, thus cutting the image and missing left and right of the picture.

**Reply by mgineer85** [reply]
Hm, it should work as follows:
Assume in Merge definition width: 1000 and height: 500. the aspect ratio is 2:1.

Now the webcam delivers an image width: 1200 and height: 500.

Result:
The left and right part of the webcam capture (100px each side) is cropped.

According to your description, it works like this and is intended.
If you do not want the images to be cropped, you have to set height: 500 and width: 1200. or multiply both values by the same number if you want to resize.

Maybe you could provide also an image?

**Comment by Mevel** [comment]
ok just to wrap this up

Thanks to @mgineer85 i got everything working

Canvas size and raw captured image size from camera sensor are different.

in the merge definition you resize or crop the camera image deliverd natively by the sensor. in order to not crop the image, you need to adjust with the correct aspect ratio (of the camera image NOT canvas size) otherwhise it will be cropped.

hope that helps other guys out there too 

cheers =)
